### ISSUE TITLE
Incorrect error formatting for generic attribute missing errors

### ISSUE THREAD

**Issue by 333fred** [dotnet/roslyn#69700]
**Version Used**: 5fcefbef

**Steps to Reproduce**:

```cs
[SomeAttribute<int>] // error CS0246: The type or namespace name 'SomeAttribute<>Attribute' could not be found (are you missing a using directive or an assembly reference?)
void Foo() {}
```
[Sharplab](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+BtAygewLYFMCCwwATgJYBGArsDgDwkB2wAfALoCwAUAAIAsABADEMGABQBKPgG8AvkA)

**Diagnostic Id**: CS0246

**Expected Behavior**:

`The type or namespace name 'SomeAttributeAttribute<>' could not be found (are you missing a using directive or an assembly reference?)`

**Actual Behavior**:

`The type or namespace name 'SomeAttribute<>Attribute' could not be found (are you missing a using directive or an assembly reference?)`

**Comment by 333fred** [comment]
Important note: there is no issue with the actual check. `SomeAttributeAttribute<T>` resolves correctly. It's just the error message that's wonky.

**Comment by kinkeadian25** [comment]
Hi, been a bit to work on anything with roslyn cause I was getting up to speed on my new job, but ready to start working on my favorite project again! @CyrusNajmabadi or @333fred. Is it cool if I take this one? I think I have narrowed it down to this snippet in Binder_Symbols.cs

if (options.IsAttributeTypeLookup() && !options.IsVerbatimNameAttributeTypeLookup())
{
    // just recurse one level, so cheat and OR verbatim name option :)
    NotFound(where, simpleName, arity, whereText + "Attribute", diagnostics, aliasOpt, qualifierOpt, options | LookupOptions.VerbatimNameAttributeTypeOnly);
}

**Comment by CyrusNajmabadi** [comment]
> Is it cool if I take this one?

Totally :-)
